cfpb/sbl-frontend
# (Review first) [Page] Update Institution profile - Phase 1,2,3

Phase 1-3 of #222 

## Changes
- Creates `Update your financial institution profile` page
- Presents all of the associated input fields
- Populates input fields with pre-existing Institution data
- Upon `Submit`, simulates a successful submission and displays the corresponding `/summary` confirmation page

## Pending issues (tracked in #222)
- [ ] Bug: Page wrapper does not match whatever we use for `View Institution...` causing a noticeable shift in page content when navigating between `View...` and `Update...`.
- [ ] TypeScript: Revisit disabled linter errors
- [ ] Feat: Integrate w/ Mail API
- [x] Fixed in #282
    - [x] Bug: `Clear form` button doesn't work, need to refactor the way we generate `defaultValues`
    - [x] Bug: Default `checked` for checkboxes does not register as in the DTO.  May be fixed with the ^ above ^ `defaultValues` refactor.     
    - [x] Bug?: User can resize the `Additional details` TextArea outside of grey bounding box.  Do we need to restrict this?

## How to test this PR
1. Login with a user who has an associated institution
2. Click on the institution to `View` it's profile
3. Click on `Update your financial institution profile`
4. Modify any institution data
5. Hit `Submit`
6. Verify you are redirected to the `/summary` route and shown a simulated "Success" confirmation

## Screenshots
### Form
![screencapture-localhost-8899-institution-TESTBANK123-update-2024-02-23-13_38_26](https://github.com/cfpb/sbl-frontend/assets/2592907/ac8d589e-d59e-4949-9e00-8186ae0355cd)

### (Simulated) Submission confirmation
![screencapture-localhost-8899-summary-2024-02-23-13_38_41](https://github.com/cfpb/sbl-frontend/assets/2592907/942b685d-5d52-4850-b2a4-8a2bcd790985)
